programs/ssl/ssl_client2.c

```diff
@@ -1682,6 +1685,7 @@ int main( int argc, char *argv[] )
         }
         else if( strcmp( p, "query_config" ) == 0 )
         {
+            opt.query_config_mode = 1;
```

Comment:
What about to introduce a "query_config_ret" local variable and thus not change anything with the usage of `ret` here and below. Then at the end of the `main()` function check `opt.query_config_mode` and if it is different from `DFL_QUERY_CONFIG_MODE` call `mbedtls_exit(  query_config_ret )`.